hooks/pre_prompt.py

```diff
@@ -0,0 +1,45 @@
+import os
+import json
+import requests
+
+def get_latest_version(package_name):
+    """Fetch the latest version of a package from PyPI."""
+    url = f"https://pypi.org/pypi/{package_name}/json"
+    response = requests.get(url)
+    if response.status_code == 200:
+        data = response.json()
+        return data["info"]["version"]
+    else:
+        raise ValueError(f"Could not fetch version for {package_name}")
+
+def main():
+    """Update cookiecutter context dynamically with current package versions."""
+    # Fetch versions for the required packages
+    packages = {
+        "vivarium": "vivarium",
+        "vivarium_public_health": "vivarium-public-health",
+        "vivarium_cluster_tools": "vivarium-cluster-tools",
+        "vivarium_inputs": "vivarium-inputs",
+        "gbd_mapping": "gbd-mapping",
+    }
+    versions = {key: get_latest_version(value) for key, value in packages.items()}
+
+    # Update the context dynamically
+    context_file = os.path.join(os.getcwd(), "cookiecutter.json")
+    with open(context_file, "r") as file:
+        context = json.load(file)
+
+    # Set default for current year if not provided
+    from datetime import datetime
+    context["current_year"] = str(datetime.now().year)
+
+    # Inject the fetched versions into the context
+    for key, version in versions.items():
+        context[f"{key}_version"] = version
+
+    # Write updated context back to cookiecutter.json
+    with open(context_file, "w") as file:
+        json.dump(context, file, indent=4)
```

Comment:
I think this is appending to the bottom of the file, correct? I'd recommend deleting the five version lines from cookiecutter.json. Otherwise (I think) you end up w/
```
    "vivarium_version": "",
    "vivarium_public_health_version": "",
    "vivarium_cluster_tools_version": "",
    "vivarium_inputs_version": "",
    "gbd_mapping_version": ""
    "vivarium_version": "1.2.3",
    "vivarium_public_health_version": "1.2.3",
    "vivarium_cluster_tools_version": "1.2.3",
    "vivarium_inputs_version": "1.2.3",
    "gbd_mapping_version": "1.2.3"
```

Reply:
Since the file is opened with "w", it is overwriting the original.  Here is the before and after

```
(base) abie@LT-MJ0L9VRK:~/vivarium_research_template$ cat cookiecutter.json
{
    "package_name": "",
    "current_year": "",
    "package_version": "0.0.1",
    "package_description": "Vivarium simulation model for the {{cookiecutter.package_name}} project.",
    "package_url": "https://github.com/ihmeuw/{{cookiecutter.package_name}}",
    "ssh_url": "git@github.com:ihmeuw/{{cookiecutter.package_name}}.git",
    "vivarium_version": "",
    "vivarium_public_health_version": "",
    "vivarium_cluster_tools_version": "",
    "vivarium_inputs_version": "",
    "gbd_mapping_version": ""
}
(base) abie@LT-MJ0L9VRK:~/vivarium_research_template$ python hooks/pre_prompt.py
(base) abie@LT-MJ0L9VRK:~/vivarium_research_template$ cat cookiecutter.json
{
    "package_name": "",
    "current_year": "2024",
    "package_version": "0.0.1",
    "package_description": "Vivarium simulation model for the {{cookiecutter.package_name}} project.",
    "package_url": "https://github.com/ihmeuw/{{cookiecutter.package_name}}",
    "ssh_url": "git@github.com:ihmeuw/{{cookiecutter.package_name}}.git",
    "vivarium_version": "3.2.10",
    "vivarium_public_health_version": "3.1.3",
    "vivarium_cluster_tools_version": "2.1.1",
    "vivarium_inputs_version": "5.2.2",
    "gbd_mapping_version": "4.1.2"
```